internal/k8s/controller.go

```diff
@@ -905,6 +909,10 @@ func (c *AviController) SetupEventHandlers(k8sinfo K8sinformers) {
 				return
 			}
 			key := utils.Ingress + "/" + utils.ObjKey(ingress)
+			// optimization to check if ingress belongs to ingressClass handled by AKO.
+			if !lib.ValidateIngressForClass(key, ingress) {
+				return
+			}
```

Comment:
dont you need something similar for the updatefunc below ?

Reply:
Also not doing anything for delete seems like an issue, how do you handle the case where an ingress that was being managed by AKO, someone then adds an ingress class later. In that case the conrresponding items should be deleted from Avi , right ?

Reply:
The ingressClass validation is not added in the update function to handle the transition of ingress from non-avi managed to avi managed and vice versa.

Reply:
With this optimization, AKO ignores the ingresses that are part of non-avi managed ingressClass.